Jormun: Admin to admin in experimental

antoine-de opened the pull request:
implement the admin to admin journeys in experimental.

we can start(finish) from(to) either 
 * the stop points of the admin main stop areas at a cost of 0 
 * or we can walk from the center of the admin

A crow fly section is created from the admin to the stop points

linked to http://jira.canaltp.fr/browse/ITI-279

The tests will be done in artemis

TODO:
- [x] wait until some artemis tests are done locally

antoine-de committed:
fill main_stop_areas

antoine-de committed:
admin to admin fully working

antoine-de committed:
add comment

antoine-de committed:
submodule bad sha1

TeXitoi commented on source/jormungandr/jormungandr/scenarios/experimental.py:
```diff
@@ -30,6 +30,7 @@
 from __future__ import absolute_import, print_function, unicode_literals, division
 import logging
 from jormungandr.scenarios import new_default
+from jormungandr.scenarios.default import is_admin
```
useless?

TeXitoi commented on source/type/pb_converter.cpp:
```diff
@@ -429,6 +429,14 @@ void PbCreator::Filler::fill_pb_object(const ng::Admin* adm, pbnavitia::Administ
+            auto* pb_sp = pb_sa->add_stop_points();
+            fill_pb_object(sp, pb_sp);
+        }
+    }
```
It can be great if we limit the size of that to the bare minimum. depth=0 for the 2 I think.

antoine-de replied:
yes you're right!

TeXitoi reviewed (commented):
Great!

antoine-de committed:
corrections after review

pbougue commented on source/jormungandr/jormungandr/scenarios/experimental.py:
```diff
@@ -97,8 +97,25 @@ def create_parameters(request):
 
-def _update_crowfly_duration(instance, mode, stop_area_uri, requested_point):
-    stop_points = instance.georef.get_stop_points_for_stop_area(stop_area_uri)
+def _update_crowfly_duration(instance, mode, requested_point):
```
`requested_place` maybe

TeXitoi replied:
I suspect that's `requested_(entry_)point`

pbougue commented on source/jormungandr/jormungandr/scenarios/experimental.py:
```diff
@@ -118,7 +135,7 @@ def _update_crowfly_duration(instance, mode, stop_area_uri, requested_point):
             odt_stop_points.add(stop_point.uri)
-    
+
     return crowfly_sps, odt_stop_points, fallback_list
```
@TeXitoi this is always free, so I see no point in `fallback_list` here

TeXitoi replied:
I don't understand your remark.

pbougue replied:
The time is always 0 in fallback

TeXitoi replied:
https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/scenarios/experimental.py#L502

That's the same format as `g.*_places_crowfly`

pbougue replied:
ok, that's fine as it is

pbougue reviewed (approved):
Adding tests would be nice, but I can't imagine interesting ones

antoine-de committed:
dump main stop only for depth > 0

antoine-de committed:
dump main stop only for depth = 2

to avoid performance issues, we dump the main stops of an admin only for
depth=2

and now place_uri query with depth = 2

antoine-de commented:
artemis tests done locally
